src/data_info/services/entry.clj

```diff
@@ -166,13 +184,24 @@
      :folders (mapv xformer collections)}))
 
 
-(defn- is-bad?
-  "Returns true if the map is okay to include in a directory listing."
-  [bad-indicator path]
-  (let [basename (file/basename path)]
-    (or (contains? (:paths bad-indicator) path)
-        (contains? (:names bad-indicator) basename)
-        (not (duv/good-string? (:chars bad-indicator) basename)))))
+(defn- file-exists-under-path
+  [irods user zone path bad-indicator filename]
+  (when (= @(rods/object-type irods user zone (file/path-join path filename)) :file)
+    (fmt-entry irods user zone (file/path-join path filename) bad-indicator)))
+
+(defn- has-readme?
+  [irods user zone path bad-indicator & maybe-listing]
+  (let [check-file (partial file-exists-under-path irods user zone path bad-indicator)]
+    (delay
+      (when maybe-listing (force maybe-listing))
+      (or
+        (check-file "README.md")
+        (check-file "README.txt")
+        (check-file "README")
+        (check-file "readme.md")
+        (check-file "readme.txt")
+        (check-file "readme")
+        false))))
```

Comment:
Minor, but maybe these could be renamed a little now that the full file stat is returned for a readme, rather than just true/false:
```suggestion
(defn- find-readme-under-path
  [irods user zone path bad-indicator filename]
  (when (= @(rods/object-type irods user zone (file/path-join path filename)) :file)
    (fmt-entry irods user zone (file/path-join path filename) bad-indicator)))

(defn- readme-stat
  [irods user zone path bad-indicator & maybe-listing]
  (let [find-readme (partial find-readme-under-path irods user zone path bad-indicator)]
    (delay
      (when maybe-listing (force maybe-listing))
      (or
        (find-readme "README.md")
        (find-readme "README.txt")
        (find-readme "README")
        (find-readme "readme.md")
        (find-readme "readme.txt")
        (find-readme "readme")
        false))))
```
Or something like this 🤷

Reply:
That's a good idea. I'll make a change like this before merging.